nb2/bot/slackbot.py

```diff
@@ -1,23 +1,174 @@
+import re
+from collections import namedtuple
+
 from slack import WebClient
 from slackeventsapi import SlackEventAdapter
 
+from nb2.service.dtos import AddQuoteDTO, CreatePersonDTO
+from nb2.service.person_service import create_person, get_person_by_slack_user_id
+from nb2.service.quote_service import add_quote_to_person
+from nb2.service.slack_service import (
+    get_quote_content_from_remember_command,
+    get_user_ids_from_command,
+)
+
 
 class SlackBot:
+    """
+    A class representing Nostalgiabot's interface with Slack.
+    """
+
+    # Container used by Actions to group returned data
+    # Fields:
+    #   ok: (bool) True if the Action succeeded. False otherwise.
+    #   message: (str) text to send back to the Slack client.
+    Result = namedtuple("Result", ["ok", "message"])
+
     def __init__(self):
         self.web_client = None
         self.event_adapter = None
+        self.slack_user_id = None
 
     def init_app(self, token, signing_secret, event_url, app):
         self.web_client = WebClient(token=token)
         self.event_adapter = SlackEventAdapter(signing_secret, event_url, app)
 
+    def init_bot_slack_user_id(self):
```

Comment:
Do you think this belongs as a private method? I'm not sure if the intention is just to call it via `get_bot_slack_user_id`?

Reply:
Yeah seems like it at this point